libraries/chain/block_state.cpp

```diff
@@ -103,10 +103,13 @@ block_state_ptr block_state::create_if_genesis_block(const block_state_legacy& b
 
    // build leaf_node and validation_tree
    valid_t::finality_leaf_node_t leaf_node {
-      .block_num       = bsp.block_num(),
-      .finality_digest = result.strong_digest,
-      .action_mroot    = *bsp.action_mroot_savanna
+      .block_num        = bsp.block_num(),
+      .timestamp        = bsp.timestamp().to_time_point(),
+      .parent_timestamp = block_timestamp_type().to_time_point(), // for the genesis block, the parent_timestamp is the the earliest representable timestamp.
```

Comment:
I think both of these should be `block_timestamp_type` instead of `time_point`. `fc::time_point` is a `int64_t` where `block_timestamp_type` is a `uint32_t`. Also seems more reflective of what this is. It is a block time stamp not a time point.

Reply:
<img width="356" alt="image" src="https://github.com/user-attachments/assets/898cce7b-d51a-4779-afa1-45a01c2c2361">

Reply:
@systemzax I believe this should be block_timestamp_type, what serialization issues does it cause?

Reply:
Originally, @greg7mdp had them as block_timestamp_type, but we don't really have an equivalent in CDT, do we?

If I'm going to work with time_point on the CDT side, it seems easier to serialize them this way already, rather than have to create a custom serializer on the contract side?

Reply:
There is `eosio::block_timestamp` in `<eosio/time.hpp>`

Reply:
Oh somehow I missed that! Thanks, will use this instead

Reply:
Addressed here : https://github.com/AntelopeIO/spring/pull/396/commits/db6248f1e170314b2fd95f98fd906a66d05ecf50

Reply:
@heifner I reverted to timestamps in the PR and disabled the `svnn_ibc` tests. I think the PR is in a state where it can be merged.